### PR TITLE
Add code to verify secrets before deployment and validate secret usages

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Review by MattGal — APPROVED** [review]
could use a comment or link to docs.

**Comment by alexperovich** [comment]
Once I get the usage stabilized and docs written I will put comments here.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -149,6 +149,14 @@ stages:
           name: Build
           displayName: Build / Publish
           condition: succeeded()
+        
+        - powershell: |
+            $manifestArgs = @()
+            Get-ChildItem .vault-config/*.yaml |% {
+              $manifestArgs += @("-m", $_.FullName)
+            }
+            dotnet run -p src/Microsoft.DncEng.SecretManager -- -b src @manifestArgs
```
Comment (jakubstilec) [inline azure-pipelines.yml:158]: Isn't it missing validate-all command?

**Review by jakubstilec — APPROVED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (alexperovich) [reply azure-pipelines.yml:158]: Yea, thats missing. Good catch.

**Comment by riarenas** [comment]
The new job was stuck waiting for permissions to the new secret manager service connection. Once I granted permissions, the job failed: https://dev.azure.com/dnceng/internal/_build/results?buildId=1222067&view=results

**Comment by MattGal** [comment]
Looks like it could just use some:
``` 
    - task: UseDotNet@2
      displayName: 'Install .NET Core 3.1.x'
      inputs:
        packageType: sdk
        version: 3.1.x
        installationPath: $(Agent.ToolsDirectory)/dotnet
```

**Comment by alexperovich** [comment]
https://github.com/dotnet/arcade-services/pull/1670

**Pull request merged**